File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.js
/**
 * @param {string[][]} items
 * @param {string} ruleKey
 * @param {string} ruleValue
 * @return {number}
 */
var countMatches = function (items, ruleKey, ruleValue) {
    let index;
    let count = 0
    switch (ruleKey) {
        case 'type':
            index = 0;
            break;
        case 'color':
            index = 1; break;
        case 'name':
            index = 2;
            break;
        default:
            index = 0;
    }

    for (let i = 0; i < items.length; i++) {
        if (items[i][index] === ruleValue) {
            count++
        }
    }
    return count
};

File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.py
class Solution:
    def countMatches(self, items: List[List[str]], ruleKey: str, ruleValue: str) -> int:
        index=0
        answer=0
        if ruleKey=='type':
            index=0
        elif ruleKey=='color':
            index=1
        else:
            index=2

        for item in items:
            if item[index]==ruleValue:
                answer+=1
        return answer
        